lib/std.dl

```diff
@@ -58,6 +58,10 @@ extern function string_join(strings: Vec<string>, sep: string): string
 extern function string_split(s: string, sep: string): Vec<string>
 extern function string_contains(s1: string, s2: string): bool
 
+/* Substring of `s` that begins at `start` and continues up to, but not
+ * including, `end`.*/
+extern function string_slice(s: string, start: bit<64>, end: bit<64>): string
```

Comment:
slice in python, javascript, etc. means something else. You should really call this substring or substr.
Also, why 64-bit indexes?